app/policies/request_policy.rb

```diff
@@ -32,13 +37,43 @@ def show?
     resource_check('read')
   end
 
-  # only for child requests
-  def index?
-    resource_check('read', record.id, record.class)
+  def user_capabilities
+    super.merge(valid_actions_hash)
+  end
+
+  def valid_actions_hash
+    hash = { Action::APPROVE_OPERATION => false,
+             Action::CANCEL_OPERATION  => false,
+             Action::DENY_OPERATION    => false,
+             Action::MEMO_OPERATION    => true }
+
+    actions = if admin?(record.class)
```

Comment:
Which api will return capabilities - get, list, or both? Here you only consider the login user has single role, but he can be a combination of all roles. For list the role is selected by persona header; for get we probably need to combine all actual roles.

Reply:
@bzwei Its for list and show calls, anytime we fetch an object these attributes will be added. The object definition has been updated in the openapi.json

Reply:
@bzwei I think the user's capabilities shouldn't set by checking persona. I will add the logic to deal with multiple roles combination